### ISSUE TITLE
Complete example fails plan with two service integrations

### ISSUE THREAD

**Issue by jademackay** [terraform-aws-modules/terraform-aws-step-functions#15]
## Description

Terraform plan on a pared down version of the complete example (`terraform-aws-step-functions/examples/complete/`) returns ` Error: Inconsistent conditional result types`

## Reproduction
```
cd terraform-aws-step-functions/examples/complete
terraform init
```
Edit content of main.tf as follows (commented such that only lambda and dynamodb remain):
```
...
  service_integrations = {

    dynamodb = {
      dynamodb = ["arn:aws:dynamodb:eu-west-1:052212379155:table/Test"]
    }

    # athena_StartQueryExecution_Sync = {
    #   athena        = ["arn:aws:athena:eu-west-1:123456789012:something1:test1"]
    #   glue          = ["arn:aws:glue:eu-west-1:123456789012:something2:test1"]
    #   s3            = true # options: true (use default value from `aws_service_policies`) or provide a list of ARNs
    #   lakeformation = ["arn:aws:lakeformation:eu-west-1:123456789012:something3:test1"]
    # }

    lambda = {
      lambda = ["arn:aws:lambda:eu-west-1:123456789012:function:test1", "arn:aws:lambda:eu-west-1:123456789012:function:test2"]
    }

    # xray = {
    #   xray = true
    # }

    #    # NB: This will "Deny" everything (including logging)!
    #    no_tasks = {
    #      deny_all = true
    #    }
  }
...
```
Resulting error:
```
$ terraform-aws-step-functions/examples/complete/ master* terraform plan
╷                                                                                        
│ Error: Inconsistent conditional result types
│ 
│   on ../../main.tf line 76, in data "aws_iam_policy_document" "service":
│   76:   for_each = local.create_role && var.attach_policies_for_integrations ? var.service_integrations : tomap({})
│     ├────────────────
│     │ var.service_integrations is object with 2 attributes
│ 
│ The true result value has the wrong type: element types must all match for conversion
│ to map.
╵
```

## Versions
 - Terraform: Terraform v0.15.0
 - Provider(s): terraform providers -version :
+ provider registry.terraform.io/hashicorp/aws v3.39.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
 - Module: terraform-aws-step-functions

I would appreciate any suggestions!

**Comment by antonbabenko** [comment]
Thanks for opening this issue! I have just fixed it in #16.

[v2.1.0](https://github.com/terraform-aws-modules/terraform-aws-step-function/releases/v2.1.0) has been just released.

**Comment by jademackay** [comment]
Thank you!
